src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/DocumentControllerTest.java

```diff
@@ -178,16 +189,28 @@ void shouldDownloadDocumentFromDocumentManagement_FromCaseDocumentClientApi() th
             .getDocumentBinary(anyString(), anyString(), eq(documentId));
     }
 
+    @Test
+    void shouldReturnExpectedGeneratedDocument() throws Exception {
+        CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted()
+            .legacyCaseReference(REFERENCE_NUMBER)
+            .totalClaimAmount(BigDecimal.ONE)
+            .issueDate(DATE)
+            .build();
+        when(restTemplate.exchange(anyString(), eq(HttpMethod.POST), any(), eq(byte[].class)))
+            .thenReturn(ResponseEntity.of(Optional.of(bytes)));
+        when(caseDocumentClientApi.uploadDocuments(anyString(), anyString(), any()))
+            .thenReturn(new UploadResponse(List.of(document)));
+
+        MvcResult result = doPost(BEARER_TOKEN, caseData, GENERATE_DOC_URL)
+            .andExpect(status().isOk()).andReturn();
+
+        JSONObject jsonReturnedCaseDocument = new JSONObject(result.getResponse().getContentAsString());
+        assertEquals(FILE_NAME, jsonReturnedCaseDocument.get("documentName"),
+                     "Document file names should match");
```

Comment:
Put a `// Then` comment in this section.

```suggestion
        // Then: the response should contain the expected document name
        JSONObject jsonReturnedCaseDocument = new JSONObject(result.getResponse().getContentAsString());
        assertEquals(FILE_NAME, jsonReturnedCaseDocument.get("documentName"),
                     "Document file names should match");
```